pvlib/pvsystem.py

```diff
@@ -1778,9 +1791,48 @@ def singlediode(photocurrent, saturation_current, resistance_series,
 
     Notes
     -----
-    The solution employed to solve the implicit diode equation utilizes
-    the Lambert W function to obtain an explicit function of V=f(i) and
-    I=f(V) as shown in [2].
+    The default method employed is an explicit solution using [4] to find an
+    arbitrary point on the IV curve as a function of the diode  voltage
+    :math:`V_d = V + I*Rs`. Then the voltage is backed out from :math:`V_d`.
+    A specific desired point, such as short circuit current or max power, is
+    located using the bisection search method, ``brentq``, bounded by a zero
+    diode voltage and an estimate of open circuit voltage given by
+
+    .. math::
+
+        V_{oc, est} = n Ns V_{th} \log \left( \frac{I_L}{I_0} + 1 \right)
+
+    We know that :math:`V_d = 0` corresponds to a voltage less than zero, and
+    we can also show that when :math:`V_d = V_{oc, est}`, the resulting
+    current is also negative, meaning that the corresponding voltage must be
+    in the 4th quadrant and therefore greater than the open circuit voltage
+    (see proof below). Therefore the entire forward-bias 1st quadrant IV-curve
+    is bounded, and a bisection search within these points will always find
+    desired condition.
+
+    .. math::
+
+        I = I_L - I_0 \left( \exp \left( \frac{V_{oc, est} }{ n Ns V_{th} } \right) - 1 \right) - \frac{V_{oc, est}}{R_{sh}} \\
+        I = I_L - I_0 \left(\exp \left( \frac{ n Ns V_{th} \log \left( \frac{I_L}{I_0} + 1 \right) }{ n Ns V_{th} } \right) - 1 \right) - \frac{n Ns V_{th} \log \left( \frac{I_L}{I_0} + 1\right)}{R_{sh}} \\
+        I = I_L - I_0 \left(\exp \left( \log \left( \frac{I_L}{I_0} + 1 \right) \right)  - 1 \right) - \frac{n Ns V_{th} \log \left( \frac{I_L}{I_0} + 1\right)}{R_{sh}} \\
+        I = I_L - I_0 \left(\frac{I_L}{I_0} + 1  - 1 \right) - \frac{n Ns V_{th} \log \left( \frac{I_L}{I_0} + 1\right)}{R_{sh}} \\
+        I = I_L - I_0 \left(\frac{I_L}{I_0} \right) - \frac{n Ns V_{th} \log \left( \frac{I_L}{I_0} + 1\right)}{R_{sh}} \\
+        I = I_L - I_L - \frac{n Ns V_{th} \log \left( \frac{I_L}{I_0} + 1\right)}{R_{sh}} \\
+        I = - \frac{n Ns V_{th} \log \left( \frac{I_L}{I_0} + 1\right)}{R_{sh}}
+
+    If ``method.lower() == 'newton'`` then a gradient descent method, ``newton``,
+    is used to solve the implicit diode equation. It should be safe for well
+    behaved IV-curves, but the default method is recommended for reliability,
+    it is often just as fast.
+
+    If either the "newton" or "brentq" methods are indicated, then
+    :func:`pvlib.pvsystem.bishop88` is used to calculate the points at diode
+    voltages from zero to open-circuit voltage with a log spacing so that
+    points get closer as they approach the open-circuit voltage.
+
+    If ``method.lower() == 'lambertw'`` then the solution employed to solve the
```

Comment:
`lambertw` is the default, so move this paragraph up in the order.

Reply:
✔️